substrate/frame/revive/src/vm/evm/instructions/contract.rs

```diff
@@ -193,24 +184,36 @@ pub fn static_call<E: Ext>(interpreter: &mut Interpreter<E>) -> ControlFlow<Halt
 fn run_call<'a, E: Ext>(
 	interpreter: &mut Interpreter<'a, E>,
 	callee: H160,
+	gas_limit: U256,
 	input: Vec<u8>,
 	scheme: CallScheme,
-	gas_limit: Weight,
 	value: U256,
 	return_memory_range: Range<usize>,
 ) -> ControlFlow<Halt> {
+	// We use ALL_STIPEND to detect the typical gas limit solc defines as a call stipend
+	// This is just a heuristic
+	let add_stipend =
+		!value.is_zero() || gas_limit.try_into().is_ok_and(|limit: u64| limit == CALL_STIPEND);
```

Comment:
I think those should be two conditionals:
1) If `value !=0`: Set `gas_limit` to at least 2300 (no heuristic just EVM semantics). Not all value transfers are plain transfers. Could call into more complicated logic. Don't think we can mess with disabling reentrancy here.
2) If `limit == CALL_STIPEND`: Only this is our heuristic. Transfer detected. Disable reentrancy but do not touch the `weight_limit`.